src/Components/Panels/Inventory/ObstacleSummary.tsx

```diff
@@ -78,13 +93,76 @@ function getSummary(obstacle: IBaseObject): JSX.Element {
 
 const ObstacleSummary = forwardRef<HTMLDivElement, Props>(
   ({ obstacle, onClick, selected }: Props, ref) => {
+    const [world, setBZWDocument] = useRecoilState(documentState);
+    const selectedUUID = useRecoilValue(selectionState);
+    const [nameEdit, setNameEdit] = useState(obstacle.name ?? '');
+    const [editMode, setEditMode] = useState(false);
+    const inputRef = useRef<HTMLInputElement>(null);
+
+    useEffect(() => {
+      if (editMode) {
+        inputRef.current!.focus();
+      } else {
+        inputRef.current!.blur();
+      }
```

Comment:
```suggestion
      if (editMode) {
        inputRef.current?.focus();
      } else {
        inputRef.current?.blur();
      }
```

Try to avoid using `!` unless you're _absolutely_ sure it'll exist. Otherwise, it's safer to lean on `?` since it won't throw an error if it's non-existent; it'll just fail gracefully.